### DISCUSSION TITLE
Some images don't look clear enough when converting from pdf to png

### DISCUSSION THREAD

**Discussion by EL-bm** [dlemstra/Magick.NET#739] [Help] [answered]
Hello,
I am currently working with the following set:
Magick.Net-Q16-HDRI-x64
Ghostscript 9.53.1
.NET Framework 4.5

I was able to read and convert a PDF file into a PNG format and extract tiles. 
Like this:
`MagickNET.SetGhostscriptDirectory(ghostscriptPath);`
`var settings = new MagickReadSettings();`
`    settings.Density = new Density(300);`
`using (var images = new MagickImageCollection())`
`{`
`    // Read all the pages of the pdf to the collection. `
`    images.Read(_stream`
`        , settings);`
    `foreach(var image in images)`
    `{`
`        image.Alpha(Alpha.Options.Remove);`
`        image.Format = MagickFormat.Png00;`
 `       image.Quality = 100;`
`        // image.Resize()`
`        image.RePage();`
`        // Extract tiles.`
`        foreach (MagickImage _tile in _levelImage.CropToTiles(512,`
`            512))`
`        {`
`            // Here I save the tiles.`
`        }`
`    }`
`}`

Yet, I haven't found the right combination for certain files. Here is a sample file:
[00030000 CG1 14 CAL GREEN.pdf](https://github.com/dlemstra/Magick.NET/files/5263099/00030000.CG1.14.CAL.GREEN.pdf)

Please pay special attention to the tables at the left of the document. The lines and the text on these tables are loosing a lot of quality on the conversion process.

Here some extracted tiles to illustrate the result:
Png 1:
![0_6](https://user-images.githubusercontent.com/71720005/93918213-a3852d80-fcd1-11ea-919a-c9875e797a33.png)

Png 2:
![1_6](https://user-images.githubusercontent.com/71720005/93918227-a718b480-fcd1-11ea-83b4-3cb6ad07d5d4.png)

Png 3:
![2_6](https://user-images.githubusercontent.com/71720005/93918238-aa13a500-fcd1-11ea-9057-75cd3100eba9.png)

In this png files you can appreciate how the table lines are not super crisp and get cutout a lot. I was able to get a better conversion using Acrobat Reader exporting the file to a PNG using the Anti-Aliasing param to "Type Optimized Hinted".

I have tried setting different **color schemas**, **compression**, **quality**, **anti-aliasing** (regular, text, and stroke), etc..., with no luck. At the end I need to be able to get something like the following image (quality-wise) using Magick.Net and Ghostscript:

Png 4 (please note the table looks more "complete"):
![3 00030000 CG1 9 CAL GREEN](https://user-images.githubusercontent.com/71720005/93919139-e693d080-fcd2-11ea-8b6a-b1701fc32185.png)

Any help will be more than appreciated!!

**Comment by dlemstra** [comment]
The problem with your PDF file is that for some reason that part of the PDF file is an image. And when Ghostscript resizes the image you appear to get a low quality result.

After some research it looks like with an option that will allow you to get a better quality image but there is no way to pass that Ghostscript option now so I will need to figure out how we can make that possible. With that extra option I get the following output:

![afbeelding](https://user-images.githubusercontent.com/10426229/93930043-094ad700-fd1d-11ea-8641-1f8a90ba34c6.png)

Will need to talk with the rest of the team to figure out how to make this option available to you.

**Comment by EL-bm** [comment]
Dear @dlemstra, thanks a lot for your prompt response. I know you guys are extremely busy so I hope you can get to this item in your todo list as fast as possible. This library is excellent and we have been using it for long now.
I'll wait to hear back from you.
Best regards!!

**Reply by dlemstra** [reply]
A patch has been applied to ImageMagick and I will make this option available in Magick.NET in the next release.

**Reply by dlemstra** [reply]
You will be able to do this by setting `Interpolate` property of the `PdfreadDefines` to true and passing in those defines when reading the image.

**Reply by EL-bm** [reply]
Thanks for the update!!
Waiting for the next release.
Thank you all.

**Reply by dlemstra** [reply]
A new release of Magick.NET that resolves this issue has been published.

**Reply by EL-bm** [reply]
Awesome!! I'll be testing this new release and closing this thread asap. Thanks again.

**Reply by EL-bm** [reply]
The answer is within this block, should I mark the entire block as the answer? Is there a way to mark @dlemstra 's comment as the answer directly?

**Comment by EL-bm** [comment, marked as answer]
Dear Magick.NET team,
I just finished up testing the new release and I am happy to report the issue reported on this thread has been successfully resolved.
I take this opportunity to express gratitude for your hard work and prompt response.

**Reply by EL-bm** [reply]
Well, here is the piece of code that finally fixed my issue:
`MagickNET.SetGhostscriptDirectory(ghostscriptPath);`
`var defines = new PdfReadDefines() { Interpolate = true };`    **// <--THIS DID IT**
`var settings = new MagickReadSettings();`
`settings.Density = new Density(300);`
`settings.SetDefines(defines);`    **// <-- AND THIS TOO**
`using (var images = new MagickImageCollection())`
`{`
`   // Read all the pages of the pdf to the collection.`
`   images.Read(_stream`
`   , settings);`
`   foreach(var image in images)`
`   {`
`      image.Alpha(Alpha.Options.Remove);`
`      image.Format = MagickFormat.Png00;`
`      image.Quality = 100;`
`      // image.Resize()`
`      image.RePage();`
`      // Extract tiles.`
`      foreach (MagickImage _tile in _levelImage.CropToTiles(512,`
`         512))`
`      {`
`         // Here I save the tiles.`
`      }`
`   }`
`}`
